### ISSUE TITLE
Error: LuaRocks 3.0.0 bug (please report at https://github.com/luarocks/luarocks/issues).

### ISSUE THREAD

**Issue by AHamburgler** [luarocks/luarocks#858]
C:\WINDOWS\system32>luarocks install "C:\lua\luasocket-3.0rc1-2.src.rock"

Error: LuaRocks 3.0.0 bug (please report at https://github.com/luarocks/luarocks/issues).
C:\Program Files (x86)\LuaRocks\lua\luarocks\fs\tools.lua:73: assertion failed!
stack traceback:
        [C]: in function 'assert'
        C:\Program Files (x86)\LuaRocks\lua\luarocks\fs\tools.lua:73: in function 'luarocks.fs.tools.change_dir'
        C:\Program Files (x86)\LuaRocks\lua\luarocks\fetch.lua:157: in function 'luarocks.fetch.fetch_and_unpack_rock'
        C:\Program Files (x86)\LuaRocks\lua\luarocks\cmd\build.lua:62: in function <C:\Program Files (x86)\LuaRocks\lua\luarocks\cmd\build.lua:55>
        (...tail calls...)
        C:\Program Files (x86)\LuaRocks\lua\luarocks\cmd\build.lua:155: in function 'luarocks.cmd.build.command'
        (...tail calls...)
        [C]: in function 'xpcall'
        C:\Program Files (x86)\LuaRocks\lua\luarocks\cmd.lua:444: in function 'luarocks.cmd.run_command'
        C:\Program Files (x86)\LuaRocks\luarocks.lua:36: in main chunk
        [C]: in ?

Error: LuaRocks 3.0.0 bug (please report at https://github.com/luarocks/luarocks/issues).
...ram Files (x86)\LuaRocks\lua\luarocks\fs\win32\tools.lua:102: assertion failed!
stack traceback:
        [C]: in function 'assert'
        ...ram Files (x86)\LuaRocks\lua\luarocks\fs\win32\tools.lua:102: in function 'luarocks.fs.win32.tools.delete'
        C:\Program Files (x86)\LuaRocks\lua\luarocks\util.lua:67: in function 'luarocks.util.run_scheduled_functions'
        [C]: in function 'xpcall'
        C:\Program Files (x86)\LuaRocks\lua\luarocks\cmd.lua:286: in local 'die'
        C:\Program Files (x86)\LuaRocks\lua\luarocks\cmd.lua:452: in function 'luarocks.cmd.run_command'
        C:\Program Files (x86)\LuaRocks\luarocks.lua:36: in main chunk
        [C]: in ?

**Comment by hishamhm** [comment]
@AHamburgler Hi, thank you for the report! Do you get the same error if you use the all-in-one binary?

**Comment by AHamburgler** [comment]
Did not try as I am not to sure how to install the all in one binary one.

**Comment by hishamhm** [comment]
@AHamburgler there is no installation, it's a zip containing a ready-to-use `luarocks.exe`

**Comment by AHamburgler** [comment]
I’ll test it out tomorrow and see if I get the same issue or not.

**Comment by murillow** [comment]
@hishamhm I am getting the same bug using Windows 10 x64.

How do you install the all-in-one `luarocks.exe` on Windows 10 x64? Whenever I try to run it, I get the following error:
`Error: Current directory does not exist. Please run LuaRocks from an existing directory.`

For instance:

```
C:\luarocks-3.0.0-windows-32>luarocks install luasocket
Error: Current directory does not exist. Please run LuaRocks from an existing directory.
What am I doing wrong?
```

**Comment by hishamhm** [comment]
@murillow please try this updated build: http://hisham.hm/win32/luarocks-dev-windows-32-8734df1.zip at let me know your results.

(A patch release should come soon!)

**Comment by murillow** [comment]
@hishamhm I guess some progress was made, because I am not getting the same error as before. But I still cannot use the all-in-one provided in the build. Here are my results:

```
C:\luarocks-dev-windows-32-8734df1>luarocks.exe install luasocket
Installing https://luarocks.org/luasocket-3.0rc1-2.src.rock

Error: Failed finding Lua library. You may need to configure LUA_LIBDIR.
```

I tried two different types of Lua installations on Windows: [LuaWinMulti](https://github.com/Tieske/luawinmulti), and building it from scratch using the Microsoft Visual Studio Build Tools 2017 toolchain. In both of them I set my lib, include and bin directories on my path. But perhaps I just suck using Windows.

**Comment by hishamhm** [comment]
@murillow thank you for the feedback — please try passing the `--lua-dir` flag to where your base Lua installation is; does it fix it?

We're still improving the workflow for Windows users; figuring out what works and what doesn't will help us make things more automatic in future releases.

**Comment by murillow** [comment]
@hishamhm yes! it does fix it! however, by default, the rocks are being installed in a directory called "systree" in the root of my base Lua installation (`C:\Lua\systree`). So I had to manually pass the `--tree` flag to the `install` command. Could you automate it aswell?

**Comment by AHamburgler** [comment]
@murillow How do I pass --lua-dir because im also getting the same error which is Error: Failed finding Lua library. You may need to configure LUA_LIBDIR.

**Comment by cup** [comment]
https://github.com/luarocks/luarocks/issues/905
